external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/OffsetManager.java

```diff
@@ -49,10 +51,14 @@ public OffsetManager(TopicPartition tp, long initialFetchOffset) {
         LOG.debug("Instantiated {}", this);
     }
 
-    public void add(KafkaSpoutMessageId msgId) {          // O(Log N)
+    public void ack(KafkaSpoutMessageId msgId) {          // O(Log N)
```

Comment:
The OffsetManager now maintains acked as well as emitted set. As @srdo  pointed **add** will be a little ambiguous.
I can make them more verbose if needed. Something like **addAck** and **addEmit**